packages/@storefront/flux-capacitor/src/core/store/middleware.ts

```diff
@@ -147,6 +147,24 @@ export namespace Middleware {
     };
   }
 
+  export function redirectAnalyzer() {
+    let isRedirecting = false;
+
+    return () => (next) => (action) => {
+      if (action.type === Actions.START_REDIRECT) {
+        isRedirecting = true;
+      } else if (action.type === Actions.DONE_REDIRECT) {
+        isRedirecting = false;
+      }
+
+      if (isRedirecting) {
```

Comment:
```js
if (!isRedirecting) return next(action);
```

Or in three lines if you think that reads better.

Reply:
This is more concise, will implement.